### PR TITLE
Feature/number input

### PR DESCRIPTION
**Pull request by iruzevic**
### Added
- Input number validation for allowed keys.

### CODE REVIEW

File: src/Blocks/components/form/assets/form.js [hunk 1]
```diff
@@ -1658,6 +1661,46 @@ export class Form {
...
+	 */
+	onKeyDownEvent = (event) => {
+		const formId = this.state.getFormIdByElement(event.target);
+		const field = this.state.getFormFieldElementByChild(event.target);
+		const name = field.getAttribute(this.state.getStateAttribute('fieldName'));
+
+		if (this.state.getStateElementTypeCustom(name, formId) === 'number') {
+			const allowedKeys = [
+				'Backspace',
+				'Enter',
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/form/assets/form.js:1679]: Can't you just filter things when setting state/submitting the form?
This seems a bit hacky and could maybe be bypassed with assistive technologies etc

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]

File: src/Blocks/components/form/assets/form.js (hunk 1 above)
Comment (iruzevic) [reply src/Blocks/components/form/assets/form.js:1679]: the problem is that I don't have anything to filter as if you get the value with text the value will be empty 👎 and I can't filter anything :(((

**Review by iobrado — APPROVED** [review]

**Review by dadadavorin — APPROVED** [review]

**Review by goranalkovic-infinum — APPROVED** [review]

File: src/Blocks/components/form/assets/form.js [hunk 2]
```diff
@@ -1658,6 +1661,46 @@ export class Form {
...
+	 *
+	 * @returns {void}
+	 */
+	onKeyDownEvent = (event) => {
+		const formId = this.state.getFormIdByElement(event.target);
+		const field = this.state.getFormFieldElementByChild(event.target);
+		const name = field.getAttribute(this.state.getStateAttribute('fieldName'));
+
+		if (this.state.getStateElementTypeCustom(name, formId) === 'number') {
+			const allowedKeys = [
```
Comment (piqusy) [inline src/Blocks/components/form/assets/form.js:1677]: How would this hold against the [Chinese numerals](https://en.wikipedia.org/wiki/Chinese_numerals) ?

Comment (iruzevic) [reply src/Blocks/components/form/assets/form.js:1677]: I have no idea really and no way to test this :(

Comment (piqusy) [reply src/Blocks/components/form/assets/form.js:1677]: Seems like Chrome (form > input field) does not support these.
From what I have found it seems that Arabic numbers are also a common method for numeric inputs in Chinese language

I'd say this is more than acceptable 😄

**Review by piqusy — APPROVED** [review]

**Pull request merged**